app/controllers/recipe_foods_controller.rb

```diff
@@ -0,0 +1,55 @@
+class RecipeFoodsController < ApplicationController
+  before_action :authenticate_user!, except: %i[index]
+  load_and_authorize_resource :recipe
+
+  def index
+    @recipe_foods = RecipeFood.include(:user).where(user_id: current_user.id)
+  end
+
+  def show
+    @recipe_food = RecipeFood.find(params[:id])
+  end
+
+  def new
+    @recipe = Recipe.find(params[:recipe_id])
+    @recipe_food = RecipeFood.new
+  end
```

Comment:
- [ ] There is one big problem in your app. on the recipe details page, when I try to add an ingredient, I get to choose the ingredients only from the foods that I have chosen before. I think here the user is free to add new ingredients other than he added before. so thats one correction you need to make as per requriements.
 
<img width="1108" alt="Screenshot 2023-08-04 at 10 27 44 PM" src="https://github.com/Jeddaa/recipe-blog/assets/96683843/2f37f6e5-f15e-4a4a-8073-f93df270b6a3">